lib/Conversion/TritonToStructured/TritonToStructuredPass.cpp

```diff
@@ -84,7 +85,8 @@ class TritonToStructuredPass
     converter.addConversion([context](RankedTensorType tensorType,
                                       SmallVectorImpl<Type> &types)
                                 -> std::optional<LogicalResult> {
-      if (!isa<triton::PointerType>(tensorType.getElementType())) {
+      if (!isa<triton::PointerType>(tensorType.getElementType()) &&
+          !tensorType.getElementType().isIntOrIndex()) {
```

Comment:
this is because only tensor of int or index can potentially be used to compute pointer offsets; i will add some more comments in a separate PR (just want to get this in first)